### PR TITLE
Add test for logical router port collector and refactor

### PR DESCRIPTION
**Pull request by walbertus**
Add test for logical router port and some refactors to help build the test:
* Wrap constructor that can pass client interface
* Change signature of metrics generator to not return prometheus.Metric
* Logical Router Port client build abstraction to list all logical router ports
* Extract method to build logical router port metrics

Address: #36 

Signed-off-by: William Albertus Dembo w.albertusd@gmail.com

### CODE REVIEW

File: collector/logical_router_port_collector.go [hunk 1]
```diff
@@ -26,8 +26,18 @@ type logicalRouterPortCollector struct {
 	txTotalByte     *prometheus.Desc
 }
 
-func newLogicalRouterPortCollector(apiClient *nsxt.APIClient, logger log.Logger) prometheus.Collector {
+type logicalRouterPortStatisticMetric struct {
+	LogicalRouterPort manager.LogicalRouterPort
+	Rx                *manager.LogicalRouterPortCounters
```
Comment (girikuncoro) [inline collector/logical_router_port_collector.go:31]: nit: should do the golang (or networking?) way
```suggestion
	RX                *manager.LogicalRouterPortCounters
```

File: collector/logical_router_port_collector.go [hunk 2]
```diff
@@ -26,8 +26,18 @@ type logicalRouterPortCollector struct {
...
+	Tx                *manager.LogicalRouterPortCounters
```
Comment (girikuncoro) [inline collector/logical_router_port_collector.go:32]: nit: should do the golang (or networking?) way
```suggestion
	TX                *manager.LogicalRouterPortCounters
```

File: collector/logical_router_port_collector_test.go [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+	return manager.LogicalRouterPortStatisticsSummary{}, errors.New("error")
+}
+
+func buildLogicalRouterPortResponse(id string, baseValue int64, err error) mockLogicalRouterPortResponse {
+	return mockLogicalRouterPortResponse{
+		ID:               fmt.Sprintf("%s-%s", fakeLogicalRouterPortID, id),
+		DisplayName:      fmt.Sprintf("%s-%s", fakeLogicalRouterPortDisplayName, id),
+		LogicalRouterID:  fmt.Sprintf("%s-%s", fakeLogicalRouterID, id),
+		Error:            err,
+		RxTotalBytes:     baseValue * 3,
```
Comment (girikuncoro) [inline collector/logical_router_port_collector_test.go]: line 80-85: any value in making this dynamic? why not just use constant value

File: collector/logical_router_port_collector.go (hunk 1 above)
Comment (walbertus) [reply collector/logical_router_port_collector.go:31]: I was thinking RX is not an initial, it should be considered as a single word
Is there any example on the official Go that uses Rx/RX?
I've found some conflicting type in the community repo

File: collector/logical_router_port_collector_test.go (hunk 3 above)
Comment (walbertus) [reply collector/logical_router_port_collector_test.go]: if it's using constant across all responses, the implementation might cheat?
But I agree using multiplication here is weird, removing the multiplication

File: collector/logical_router_port_collector.go (hunk 1 above)
Comment (girikuncoro) [reply collector/logical_router_port_collector.go:31]: found some with RX/TX
https://github.com/akhenakh/statgo/blob/3ae2cda264c55e1eaffaaa11f626b6c833eeff29/net_io_stats.go#L13

another one with Rx/Tx
https://github.com/bcicen/ctop/blob/a2c08d312ef9db1e7dad162c5c47981abab940a6/models/main.go#L10

I guess it doesn't really matter then, let's just stick to what we have

**Review by girikuncoro — APPROVED** [review]

**Pull request merged**